orcoursetrion/actions/github.py

```diff
@@ -45,6 +45,53 @@ def create_export_repo(course, term, description=None):
     return repo
 
 
+def rerun_studio(course, term, new_term, description=None):
+    """Run any actions needed to re-run a Studio course.
+
+    This will remove the hooks from the specified term, and then create a new
+    export repo for the ``new_term``.  It finds the repo based on
+    :py:const:`~orcoursetrion.config.ORC_STUDIO_ORG` as the organization,
+    :py:const:`~orcoursetrion.config.ORC_COURSE_PREFIX` as the prefix,
+    replacing all the dots in ``course`` and appending "-``term``".
+
+    Args:
+        course (str): Course name to be used to name repo (i.e. 6.004r)
+        term (str): Term the course is last run (i.e. 2015_Spring)
+        term (str): Term the course is expected to run again (i.e. 2018_Spring)
+        description (str): Optional description for repo to show up on github
+    Raises:
+        requests.RequestException
+        orcoursetrion.lib.GitHubUnknownError
+    Returns:
+        dict: Github dictionary of the newly created repo
+                (https://developer.github.com/v3/repos/#create)
+
+    """
+    github = GitHub(config.ORC_GH_API_URL, config.ORC_GH_OAUTH2_TOKEN)
+
+    # Find and clean up the old
+    repo_name = '{prefix}-{course}-{term}'.format(
+        prefix=config.ORC_COURSE_PREFIX,
+        course=course.replace('.', ''),
+        term=term
+    )
+    github.delete_web_hooks(config.ORC_STUDIO_ORG, repo_name)
```

Comment:
I think this should be idempotent actually, since delete will succeed even when no hooks exist.  It is also important to note the reverse, if we swap these and run the add first, and the delete fails, we potentially have a big problem with hooks firing from two versions of the same course.  It would be a weird edge case, but way worse than not firing at all in a failure mode.